### PR TITLE
Fix trial start time

### PR DESCRIPTION
**Pull request by chicm-ms**

### CODE REVIEW

File: src/nni_manager/core/nniDataStore.ts [hunk 1]
```diff
@@ -87,7 +87,19 @@ class NNIDataStore implements DataStore {
         event: TrialJobEvent, trialJobId: string, hyperParameter?: string, jobDetail?: TrialJobDetail): Promise<void> {
         this.log.debug(`storeTrialJobEvent: event: ${event}, data: ${hyperParameter}, jobDetail: ${JSON.stringify(jobDetail)}`);
 
-        return this.db.storeTrialJobEvent(event, trialJobId, hyperParameter, jobDetail).catch(
+        let timestamp: number | undefined;
```
Comment (QuanluZhang) [inline src/nni_manager/core/nniDataStore.ts]: could you add comments to explain the logic a little bit?

File: src/nni_manager/core/nniDataStore.ts [hunk 2]
```diff
@@ -272,6 +284,9 @@ class NNIDataStore implements DataStore {
                     if (record.logPath !== undefined) {
                         jobInfo.logPath = record.logPath;
                     }
+                    if (jobInfo.startTime === undefined && record.timestamp !== undefined) {
```
Comment (QuanluZhang) [inline src/nni_manager/core/nniDataStore.ts]: would be better to also add brief comment here

File: src/nni_manager/core/nniDataStore.ts [hunk 3]
```diff
@@ -87,7 +87,19 @@ class NNIDataStore implements DataStore {
         event: TrialJobEvent, trialJobId: string, hyperParameter?: string, jobDetail?: TrialJobDetail): Promise<void> {
         this.log.debug(`storeTrialJobEvent: event: ${event}, data: ${hyperParameter}, jobDetail: ${JSON.stringify(jobDetail)}`);
 
-        return this.db.storeTrialJobEvent(event, trialJobId, hyperParameter, jobDetail).catch(
+        let timestamp: number | undefined;
+        if (event === 'WAITING' && jobDetail !== undefined) {
```
Comment (yds05) [inline src/nni_manager/core/nniDataStore.ts]: can be simplied to if (event === 'WAITING' && jobDetail)

File: src/nni_manager/core/nniDataStore.ts [hunk 4]
```diff
@@ -87,7 +87,19 @@ class NNIDataStore implements DataStore {
...
+            timestamp = jobDetail.submitTime;
+        } else if (event === 'RUNNING' && jobDetail !== undefined) {
```
Comment (yds05) [inline src/nni_manager/core/nniDataStore.ts]: the same

File: src/nni_manager/core/nniDataStore.ts [hunk 5]
```diff
@@ -87,7 +87,19 @@ class NNIDataStore implements DataStore {
...
+            timestamp = jobDetail.startTime;
+        } else if (['EARLY_STOPPED', 'SUCCEEDED', 'FAILED', 'USER_CANCELED', 'SYS_CANCELED'].includes(event) && jobDetail !== undefined) {
```
Comment (yds05) [inline src/nni_manager/core/nniDataStore.ts]: the same

File: src/nni_manager/core/nniDataStore.ts (hunk 2 above)
Comment (chicm-ms) [reply src/nni_manager/core/nniDataStore.ts]: Thanks, updated.

File: src/nni_manager/core/nniDataStore.ts (hunk 1 above)
Comment (chicm-ms) [reply src/nni_manager/core/nniDataStore.ts]: Thanks, updated.

File: src/nni_manager/core/nniDataStore.ts (hunk 3 above)
Comment (chicm-ms) [reply src/nni_manager/core/nniDataStore.ts]: Thanks, updated.

File: src/nni_manager/core/nniDataStore.ts (hunk 5 above)
Comment (chicm-ms) [reply src/nni_manager/core/nniDataStore.ts]: Thanks, updated.

File: src/nni_manager/core/nniDataStore.ts (hunk 4 above)
Comment (chicm-ms) [reply src/nni_manager/core/nniDataStore.ts]: Thanks, updated.

**Pull request merged**